src/main/CMakeLists.txt

```diff
@@ -138,45 +135,70 @@ set(shadow_srcs
     utility/priority_queue.c
     utility/random.c
     utility/utility.c
-
-    main.c
 )
+add_library(shadow-c STATIC ${shadow_srcs})
+target_link_libraries(shadow-c INTERFACE
+   ${CMAKE_THREAD_LIBS_INIT} ${M_LIBRARIES} ${DL_LIBRARIES} ${RT_LIBRARIES}
+   ${IGRAPH_LIBRARIES} ${GLIB_LIBRARIES} ${PROCPS_LIBRARIES}
+   shadow-shim-helper logger shadow-remora shadow-shmem shadow-tsc)
+
+# TODO: extract -L and -l flags from the output of
+# `get_target_property(INTERFACE_LINK_LIBRARIES shadow-c
+# INTERFACE_LINK_LIBRARIES)`, and pass those through to cargo.
+get_target_property(INTERFACE_LINK_LIBRARIES shadow-c INTERFACE_LINK_LIBRARIES)
+foreach(INTERFACE_LINK_LIBRARY IN LISTS INTERFACE_LINK_LIBRARIES)
+    if(INTERFACE_LINK_LIBRARY MATCHES "(.*)/lib(.*)\(.so|.a)")
```

Comment:
Does this link to all of Shadow's dependencies like glib, procps, etc? Do we need all of this linking, since I'm guessing we don't plan to call into any of these external libraries manually from Rust?